File: 771. Jewels and Stones/solution.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function(jewels, stones) {
  let mySet = new Set();
  for (let index = 0; index < jewels.length; index++){
    let jewel = jewels[index];
    mySet.add(jewel);
  }

  let count = 0;
  for (let index = 0; index < stones.length; index++){
    let stone = stones[index];
    if (mySet.has(stone)){
      count += 1;
    }
  }

  return count;
};

File: 771. Jewels and Stones/solution.cpp
class Solution {
public:
  int numJewelsInStones(string jewels, string stones) {
    unordered_set<char> mySet;
    for (int index = 0; index < jewels.size(); index++){
      char jewel = jewels[index];
      mySet.insert(jewel);
    }

    int count = 0;
    for (int index = 0; index < stones.size(); index++){
      char stone = stones[index];
      if (mySet.count(stone) == 1){
        count += 1;
      }
    }

    return count;
  }
};